### ISSUE TITLE
Logger will not print on browser console if body html is cleared

### ISSUE THREAD

**Issue by vegegoku** [gwtproject/gwt#9570]
**GWT version: 2.8.2** 
**Browser (with version): Chrome Version 62.0.3202.89 (Official Build) (64-bit)** 
**Operating System: Ubuntu 16.04 LTS** 

---

##### Steps to reproduce

- create a new fresh project using the archetypes from tbroyer and specify the modular-webapp as the artifactId
https://github.com/tbroyer/gwt-maven-archetypes

- in the created project edit the entry point to look like the following

```java
public class App implements EntryPoint {
	
	private static final Logger LOGGER=Logger.getLogger(App.class.getName());
	
	public void onModuleLoad() {

		DomGlobal.document.body.innerHTML="";
		HTMLButtonElement button=(HTMLButtonElement)DomGlobal.document.createElement("button");
		button.textContent="Test";

		DomGlobal.document.body.appendChild(button);
		button.addEventListener("click", evt -> {
			LOGGER.log(Level.INFO, "this log will NOT work");
		});
		LOGGER.log(Level.INFO, "this log will work");
	}
}
```

![log-issue](https://user-images.githubusercontent.com/5379625/32865517-294271c6-ca6c-11e7-93ce-3740c66cdd1b.gif)

**Comment by tbroyer** [comment]
Remember that GWT code runs inside an iframe, so if you clear the body, you remove the iframe and "detach" it from the browsing context. This is likely to cause many errors or strange behaviors (note how we deprecated `RootPanel.clear(boolean)` in 2.8? actually, the commit message specifically points out `console.log`: https://github.com/gwtproject/gwt/commit/3d1a3e9bcbd71de881764f4efcf5ddb4398a364e).

tl;dr: don't do this :man_shrugging:

**Comment by vegegoku** [comment]
Still, even though, there should be a convenient way to clear the body, if we cant use RootPanel clear or document body innerHtml, which are supposed to be trivial things to do, there should be an equivalent API that ease this operation, it is not like i have to to track all elements of the body and remove them one by one. and also if the  `RootPanel.clear(boolean)` is deprecated the `DomGlobal.document.body.innerHtml` is not and anyone can use it without knowing and he will spend hours debugging not knowing what is going on, there were now warning and no errors to lead me to the problem cause nd spent hours til i figured such weird behavior.

**Comment by JamesXNelson** [comment]
If you want to do this, you will want to detach anything you want to survive.  Or mark anything you want to sweep, and kill everything else.

innerHTML is easy, but incorrect.  You must at least identify the iframe you are running in and keep it (or use a linker which does not install into an iframe).

**Comment by vegegoku** [comment]
its ok to say that we should not use this, but the point is that at least there should be a warning or hint somewhere, dont you think so?

**Comment by jnehlmeier** [comment]
@vegegoku sure this would be nice, but it is not really doable. Best thing to do is to document it somewhere on gwtproject.org.
